### PR TITLE
Update Fundamentals/Index to 3.1

### LINKED ISSUE #13877: 3.1 update: ASP.NET Core fundamentals
update

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a7e622e9-c567-00b9-3788-d6a01934b314
* Version Independent ID: 8e663029-63b9-2622-f852-9f8a2ec70db3
* Content: [ASP.NET Core fundamentals](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/?view=aspnetcore-2.2&tabs=windows)
* Content Source: [aspnetcore/fundamentals/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #13877.

Here's a start, which is just a basic translation for 3.1. I'm going to look at the actual contents of this topic before marking this as "Ready for review".

@Rick-Anderson  edit:
[Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/fundamentals/?view=aspnetcore-3.1&branch=pr-en-us-17321&tabs=linux)

### CODE REVIEW

**Comment by serpent5** [comment]
@Rick-Anderson Would you mind exporting the current version of this for external review? If you're curious, feel free to take a look at it in its current state. I'm almost done...

**Comment by Rick-Anderson** [comment]
External review:

* Go to [this DropBox link](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
* In the upper right, select Donwload -> Direct download
* Click on HTML files.

File: aspnetcore/fundamentals/index.md [hunk 1]
```diff
@@ -461,7 +439,7 @@ An implementation of `IHttpClientFactory` is available for creating `HttpClient`
 * Provides a central location for naming and configuring logical `HttpClient` instances. For example, a *github* client can be registered and configured to access GitHub. A default client can be registered for other purposes.
 * Supports registration and chaining of multiple delegating handlers to build an outgoing request middleware pipeline. This pattern is similar to the inbound middleware pipeline in ASP.NET Core. The pattern provides a mechanism to manage cross-cutting concerns around HTTP requests, including caching, error handling, serialization, and logging.
 * Integrates with *Polly*, a popular third-party library for transient fault handling.
-* Manages the pooling and lifetime of underlying `HttpClientMessageHandler` instances to avoid common DNS problems that occur when manually managing `HttpClient` lifetimes.
+* Manages the pooling and lifetime of underlying `HttpClientHandler` instances to avoid common DNS problems that occur when manually managing `HttpClient` lifetimes.
```
Comment (serpent5) [inline aspnetcore/fundamentals/index.md]: I wouldn't normally change content in a < 3.0 version during an update, but I think this one's worth fixing.

File: aspnetcore/fundamentals/index.md [hunk 2]
```diff
@@ -101,56 +106,49 @@ ASP.NET Core provides the following server implementations:
...
-app.UseWebSockets(options);
-```
+[!code-csharp[](index/samples_snapshot/3.x/UseWebSockets.cs)]
 
 For more information, see <xref:fundamentals/configuration/options>.
 
 ## Environments
 
-Execution environments, such as *Development*, *Staging*, and *Production*, are a first-class notion in ASP.NET Core. You can specify the environment an app is running in by setting the `ASPNETCORE_ENVIRONMENT` environment variable. ASP.NET Core reads that environment variable at app startup and stores the value in an `IHostingEnvironment` implementation. The environment object is available anywhere in the app via DI.
+Execution environments, such as *Development*, *Staging*, and *Production*, are a first-class notion in ASP.NET Core. Specify the environment an app is running in by setting the `ASPNETCORE_ENVIRONMENT` environment variable. ASP.NET Core reads that environment variable at app startup and stores the value in an `IWebHostEnvironment` implementation. This implementation is available anywhere in an app via dependency injection (DI).
```
[the remark this answers is not in the document]
Comment (Rick-Anderson) [reply aspnetcore/fundamentals/index.md]: code fence `Development`, etc so they don't get translated and because they're code.

**Comment by serpent5** [comment]
@Rick-Anderson Great feedback, thanks.

**Comment by Rick-Anderson** [comment]
@serpent5  fantastic work

**Pull request merged**